lib/Driver/ToolChain.cpp

```diff
@@ -287,6 +287,31 @@ mergeBatchInputs(ArrayRef<const Job *> jobs,
   return false;
 }
 
+/// Remove space for -l arg values from \c Args and add to \c FrontendArgs
```

Comment:
Nit: Doc comments should preferably go into headers.

Also, the comment should try to focus on things which are not clear from the code. In this case, the function is small enough that someone can look and understand what the function is doing. The thing that is not evident from the code is "why do we have this special logic for `-l`?" So the comment should try to answer that. Here's my attempt (I haven't checked if this fits in 80 columns):

```
/// Special handling for passing down '-l' arguments.
///
/// Not all downstream tools (lldb, ld etc.) consistently accept a space between the '-l'
/// flag and its argument, so we remove the extra space if it was present in \c Args.
```

Reply:
Makes sense